pkgs/tools/networking/mailutils/don-t-use-descrypt-password-in-the-test-suite.patch

```diff
@@ -0,0 +1,59 @@
+From c9b8b625b4fddd721e058cc524386af6284631d4 Mon Sep 17 00:00:00 2001
+From: Michal Sojka <michal.sojka@cvut.cz>
+Date: Sat, 1 Apr 2023 10:59:00 +0200
+Subject: [PATCH] Don't use descrypt password in the test suite
+
+The descrypt method is not considered strong and some
+distributions (like NixOS) start compiling libxcrypt without it [1].
+
+To start using different hash methods, it is not sufficient to replace
+the password hash in testsuite/etc/passwd.in, because differently
+hashed passwords contain '$' characters (e.g.
+$y$j9T$lP8UWS9GZ4zfeTVDpmtbr1$dVozXOvWQXLQyhWGX7YyW8TIQTg8EfQLUzYPp3WCWo9)
+and the test suite replaces '$'-starting strings with values of the
+corresponding variables. Since I've not found a way to escape dollars
+for TCL's subst function, we do not hardcode the passwd hash directly
+to the file, bug generate it dynamically as replace it in the file via
+a variable.
+
+[1]: https://github.com/NixOS/nixpkgs/pull/220557
+---
+ pop3d/testsuite/lib/pop3d.exp | 1 +
+ testsuite/etc/passwd.in       | 2 +-
+ testsuite/lib/mailutils.exp   | 1 +
+ 3 files changed, 3 insertions(+), 1 deletion(-)
+
+diff --git a/pop3d/testsuite/lib/pop3d.exp b/pop3d/testsuite/lib/pop3d.exp
+index 930861d8a..3b67844d2 100644
+--- a/pop3d/testsuite/lib/pop3d.exp
++++ b/pop3d/testsuite/lib/pop3d.exp
+@@ -26,6 +26,7 @@ if ![mu_check_capability ENABLE_VIRTUAL_DOMAINS] {
+ }
+ 
+ set POP3D_ETC_DIR "$MU_DATA_DIR/etc"
++set MU_PASSWD_HASH [exec mkpasswd --method=yescrypt guessme]
```

Comment:
Question: why yescrypt in particular?

Reply:
Also, can we be sure that `mkpasswd` is always available (outside of the nix expression / for generic distributions), or should autoconf look for it?

Reply:
> Question: why yescrypt in particular?

Because I saw it at many places in https://github.com/NixOS/nixpkgs/pull/220557/files. Not sure whether something else is more appropriate.

Reply:
As for the autoconf - I'm not sure. If `mkpasswd` is not found, the testsuite fails with a reasonable error message. If we add something like this to `configure.ac`:
```autoconf
AC_PATH_PROG([MKPASSWD], [mkpasswd])
if test -z "$MKPASSWD"; then
  AC_MSG_WARN([mkpasswd not found - the testsuite may fail])
fi
```
I guess most people will ignore it. And it makes little sense to turn it to `AC_MSG_ERROR`, because `mkpasswd` is only needed for tests, which people don't need to run.

Reply:
Fair enough!